tests/components/zwave_js/test_light.py

```diff
@@ -0,0 +1,377 @@
+        "propertyName": "targetValue",
+        "metadata": {
+            "label": "Target value",
+            "max": 99,
+            "min": 0,
+            "type": "number",
+            "readable": True,
+            "writeable": True,
+            "label": "Target value",
+        },
+    }
+    assert args["value"] == 255
+
+    client.async_send_json_message.reset_mock()
+
+    # Test brightness update from value updated event
+    event = Event(
+        type="value updated",
+        data={
+            "source": "node",
+            "event": "value updated",
+            "nodeId": 39,
+            "args": {
+                "commandClassName": "Multilevel Switch",
+                "commandClass": 38,
+                "endpoint": 0,
+                "property": "currentValue",
+                "newValue": 99,
+                "prevValue": 0,
+                "propertyName": "currentValue",
+            },
+        },
+    )
+    node.receive_event(event)
+
+    state = hass.states.get(BULB_6_MULTI_COLOR_LIGHT_ENTITY)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_BRIGHTNESS] == 255
+    assert state.attributes[ATTR_COLOR_TEMP] == 370
+
+    # Test turning on with same brightness
+    await hass.services.async_call(
+        "light",
+        "turn_on",
+        {"entity_id": BULB_6_MULTI_COLOR_LIGHT_ENTITY, ATTR_BRIGHTNESS: 255},
+        blocking=True,
+    )
+
+    assert len(client.async_send_json_message.call_args_list) == 0
+
+    client.async_send_json_message.reset_mock()
+
+    # Test turning on with brightness
+    await hass.services.async_call(
+        "light",
+        "turn_on",
+        {"entity_id": BULB_6_MULTI_COLOR_LIGHT_ENTITY, ATTR_BRIGHTNESS: 129},
+        blocking=True,
+    )
+
+    assert len(client.async_send_json_message.call_args_list) == 1
+    args = client.async_send_json_message.call_args[0][0]
+    assert args["command"] == "node.set_value"
+    assert args["nodeId"] == 39
+    assert args["valueId"] == {
+        "commandClassName": "Multilevel Switch",
+        "commandClass": 38,
+        "endpoint": 0,
+        "property": "targetValue",
+        "propertyName": "targetValue",
+        "metadata": {
+            "label": "Target value",
+            "max": 99,
+            "min": 0,
+            "type": "number",
+            "readable": True,
+            "writeable": True,
+            "label": "Target value",
+        },
+    }
+    assert args["value"] == 50
+
+    client.async_send_json_message.reset_mock()
+
+    # Test turning on with rgb color
+    await hass.services.async_call(
+        "light",
+        "turn_on",
+        {"entity_id": BULB_6_MULTI_COLOR_LIGHT_ENTITY, ATTR_RGB_COLOR: (255, 76, 255)},
+        blocking=True,
+    )
+
+    assert len(client.async_send_json_message.call_args_list) == 4
+    warm_args = client.async_send_json_message.call_args_list[0][0][0]  # warm white 0
+    assert warm_args["command"] == "node.set_value"
```

Comment:
Why break out the checks and not compare dictionaries ?

Reply:
The metadata is big and not really interesting.

Reply:
I'll follow up with some fixtures or constants with different payloads.